cumulusci/tasks/salesforce/check_components.py

```diff
@@ -0,0 +1,212 @@
+                # If the task is 'deploy', collect the path
+                if task_name == "deploy":
+                    path = options.get("path")
+                    if path and path not in deploy_paths:
+                        deploy_paths.append(path)
+                # If the task is 'flow', recursively get deploy paths
+                elif task_name == "flow":
+                    flow_name = options.get("flow")
+                    if flow_name:
+                        self.logger.info(f"Recursing into flow: {flow_name}")
+                        flow_steps = self._get_plan_or_flow_steps(flow_name)
+                        deploy_paths.extend(
+                            self._get_deploy_paths_from_steps(flow_steps)
+                        )
+            elif "flow" in step:
+                flow_name = step["flow"]
+                self.logger.info(f"Recursing into flow: {flow_name}")
+                flow_steps = self._get_plan_or_flow_steps(flow_name)
+                deploy_paths.extend(self._get_deploy_paths_from_steps(flow_steps))
+            # Handle nested steps
+            elif "steps" in step:
+                nested_steps = step["steps"]
+                deploy_paths.extend(self._get_deploy_paths_from_steps(nested_steps))
+            else:
+                options = step.get("options", {})
+                path = options.get("path")
+                if path and path not in deploy_paths:
+                    deploy_paths.append(path)
+        return deploy_paths
+
+    def _collect_components_from_paths(self, full_path):
+        components = defaultdict(set)
+        self.logger.info(f"Collecting components from path: {full_path}")
+
+        with convert_sfdx_source(full_path, None, self.logger) as src_path:
+            package_xml_path = os.path.join(src_path, "package.xml")
+            if os.path.exists(package_xml_path):
+                try:
+                    source_xml_tree = metadata_tree.parse(package_xml_path)
+                    self.logger.info(f"parsing package.xml: {source_xml_tree}")
+                    for types_element in source_xml_tree.findall("types"):
+                        members = [
+                            member.text for member in types_element.findall("members")
+                        ]
+                        name = types_element.find("name").text
+                        components[name].update(members)
+
+                    response_messages = self._get_api_object_responce(
+                        package_xml_path, source_xml_tree.version.text
+                    )
+
+                    return [components, response_messages]
+
+                except ParseError as e:
+                    self.logger.error(f"Error parsing package.xml: {e}")
+                    return None
+            else:
+                self.logger.info(
+                    f"No package.xml found in {full_path}, scanning directories"
+                )
```

Comment:
It might be better to log a warning here if package.xml doesn't exist, instead of an info message. This would highlight that something might be wrong if expected files are missing.